src/components/navigation/NavMenu.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  Typography,
+  Menu,
+  Button,
+  useTheme,
+  type MenuListProps,
+  MenuItem,
+  type MenuItemProps,
+} from "@mui/material";
+import React, { useState, forwardRef } from "react";
+import { ExpandMore } from "@mui/icons-material";
+import { NavLink, NavLinkProps } from "./Navbar";
+
+type NavMenuLinkProps = MenuItemProps & NavLinkProps;
+
+const NavMenuLink = forwardRef<HTMLElement, NavMenuLinkProps>(
+  function NavMenuLink({ children, ...props }: NavMenuLinkProps, ref) {
+    const theme = useTheme();
+
+    return (
+      <MenuItem
+        ref={ref}
+        component={NavLink}
+        {...props}
+        sx={{
+          "&:hover": {
+            color: theme.palette.secondary.main,
+            borderLeft: "solid 4px",
+            borderBottom: "none",
+          },
+          "&:focus": {
+            color: theme.palette.secondary.main,
+            borderLeft: "solid 4px",
+          },
+          textDecoration: "none",
+          alignItems: "center",
+          display: "flex",
+          borderLeft: "4px solid transparent",
+          backgroundColor: { md: "none" },
+          color: theme.palette.primary.contrastText,
+        }}
+      >
+        {children}
+      </MenuItem>
+    );
+  },
+);
+
+interface NavMenuProps extends MenuListProps {
+  label: string;
+}
+
+const NavMenu = ({ label, children }: NavMenuProps) => {
+  const [anchorElement, setAnchorElement] = useState<null | HTMLElement>(null);
+  const open = Boolean(anchorElement);
+  const [menuWidth, setMenuWidth] = useState(0);
+
+  const openMenu = (e: React.MouseEvent<HTMLButtonElement>) => {
+    if (!open) {
+      setAnchorElement(e.currentTarget);
+      setMenuWidth(e.currentTarget.offsetWidth);
+    }
+  };
+
+  const closeMenu = () => {
+    setAnchorElement(null);
+  };
+
+  const theme = useTheme();
+
+  return (
+    <>
+      <Button
```

Comment:
Can improve accessibility a bit here with [aria-haspopup](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-haspopup), [aria-expanded](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-expanded) and [aria-controls](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-controls) then link to the menu with `id`.